Formula/gcc@7.rb

```diff
@@ -23,6 +23,10 @@ class GccAT7 < Formula
     sha256 cellar: :any_skip_relocation, x86_64_linux: "e653d70c78fb9bcd714a6904775dcd9c3b2e661efd633722519f347e190c7c8f"
   end
 
+  # Unsupported per https://gcc.gnu.org/gcc-7/
+  # Last release on 2019-11-14
+  deprecate! date: "2023-05-18", because: :deprecated_upstream
```

Comment:
We're usually a bit more lenient about (mainstream) compilers when it comes to deprecation policies. They're typically relatively low maintenance, and they're important for users on old machines who want to build newer software.

I won't object if you feel strongly though.

Reply:
We just deprecated [`gcc@8`](https://github.com/Homebrew/homebrew-core/pull/131138) because it no longer builds on Ventura and won't be fixed. Look's like this is the same thing: https://github.com/Homebrew/homebrew-core/pull/131132

Reply:
> We just deprecated [`gcc@8`](https://github.com/Homebrew/homebrew-core/pull/131138) because it no longer builds on Ventura and won't be fixed. Look's like this is the same thing: #131132

Yea, I was meaning to say the same thing in that PR but it was merged before I got the chance to. @fxcoudert probably has more thoughts here.

Reply:
> We just deprecated [`gcc@8`](https://github.com/Homebrew/homebrew-core/pull/131138) because it no longer builds on Ventura and won't be fixed. Look's like this is the same thing: #131132

Good enough for me 👍🏻